posthog/models/feature_flag/flags_cache.py

```diff
@@ -0,0 +1,178 @@
+"""
+Flags HyperCache for feature-flags service.
+
+This module provides a HyperCache that stores feature flags for the feature-flags service.
+Unlike the local_evaluation.py cache which provides rich data for SDKs (including cohorts
+and group type mappings), this cache provides just the raw flag data.
+
+The cache is automatically invalidated when:
+- FeatureFlag models are created, updated, or deleted
+- Team models are created or deleted (to ensure flag caches are cleaned up)
+
+Cache Key Pattern:
+- Uses team_id as the key
+- Stored in both Redis and S3 via HyperCache
+
+Configuration:
+- Redis TTL: 7 days (configurable via FLAGS_CACHE_TTL env var)
+- Miss TTL: 1 day (configurable via FLAGS_CACHE_MISS_TTL env var)
+"""
+
+from typing import Any
+
+from django.conf import settings
+from django.core.cache import cache, caches
+from django.db import transaction
+from django.db.models.signals import post_delete, post_save
+from django.dispatch import receiver
+
+import structlog
+
+from posthog.caching.flags_redis_cache import FLAGS_DEDICATED_CACHE_ALIAS
+from posthog.models.feature_flag import FeatureFlag
+from posthog.models.feature_flag.feature_flag import get_feature_flags, serialize_feature_flags
+from posthog.models.team import Team
+from posthog.storage.hypercache import HyperCache
+
+logger = structlog.get_logger(__name__)
+
+# Database configuration for flag fetching (matches local_evaluation.py pattern)
+DATABASE_FOR_FLAGS = (
+    "default"
+    if ("local_evaluation" not in settings.READ_REPLICA_OPT_IN or "replica" not in settings.DATABASES)
```

Comment:
Kind of piggy-backing on "local_evaluation" which reads from the read replica.

2 questions:
1. Should we be reading from the read replica?
2. Should we add our own "feature_flags" [entry in `READ_REPLICA_OPT_IN`](https://github.com/PostHog/charts/blob/main/argocd/posthog/values/common.prod-us.yaml#L146)?